crates/deltalake-core/src/table/state_arrow.rs

```diff
@@ -167,13 +167,33 @@ impl DeltaTableState {
             })
             .collect::<HashMap<&str, _>>();
 
+        let physical_name_to_logical_name = metadata
+            .partition_columns
+            .iter()
+            .map(|name| {
+                let physical_name = metadata
+                    .schema
+                    .field_with_name(name)
+                    .unwrap()
+                    .physical_name();
+                (physical_name, name)
+            })
+            .collect::<HashMap<&String, &String>>();
```

Comment:
small nit: `&String` is almost never what we want, as it's a pointer to a pointer. You can just use `&str` here.

```suggestion
                (physical_name, name.as_str())
            })
            .collect::<HashMap<&str, &str>>();
```